text/1005-edgedb-project.rst

```diff
@@ -0,0 +1,418 @@
+  the latter containing this declaration::
+
+      module default {
+
+      }
+
+- if the specified server version is not installed, ``edgedb server install``
+  performs the installation using the first available installation method
+  in the order of preference (unless specified explicitly with
+  ``--server-instance-type``).
+
+- if the specified or implied server instance does not exist, an attempt to
+  create it is made.
+
+- a new record in ``~/.edgedb/projects`` is created for the new project.
+
+In an existing project:
+
+- the ``edgedb.toml`` file is read and validated;
+
+- if the specified server version is not installed, ``edgedb server install``
+  performs the installation using the first available installation method
+  in the order of preference (unless specified explicitly with
+  ``--server-instance-type``).
+
+- if the specified or implied server instance does not exist, an attempt to
+  create it is made.
+
+- the record in ``~/.edgedb/projects`` is updated with the new instance name
+  if necessary.
+
+- if ``dbschema/migrations`` exists, ``edgedb migrate`` is executed to ensure
+  that the configured instance is up-to-date.
+
+Interactive mode
+----------------
+
+Here's a simulation of a proposed interactive mode for a new project::
+
+    $ edgedb project init
+    `edgedb.toml` was not found in `/home/user/work/myapp` or above.
+    Do you want to initialize a new project? [Y/n] Y
+    What type of EdgeDB instance would you like to use with this project?
+    1. Local (native)
+    2. Local (Docker)
+    3. Cloud
+    Your choice? 1
+    Specify the version of EdgeDB to use with this project [latest stable]:
+    Specify the name of EdgeDB instance to use with this project [myapp]:
+
+    [shows summary of configuration]
+
+    [asks whether to continue or restart configuration]
+
+    Creating instance `myapp`...
+
+Here's a simulation of a proposed interactive mode for a cloned project::
+
+    $ edgedb project init
+    Found `edgedb.toml` in `/home/user/work/myapp`.
+    Found no associated EdgeDB instance.
+    What type of EdgeDB instance would you like to use with this project?
+    1. Local (native)
+    2. Local (Docker)
+    3. Cloud
+    Your choice? 1
+    Specify the name of EdgeDB instance to use with this project [myapp]:
+
+    [shows summary of configuration]
+
+    [asks whether to continue or restart configuration]
+
+    Creating instance `myapp` ...
+    Running migrations ...
+
+
+edgedb project deinit
+=====================
+
+Remove association with and optionally destroy the linked EdgeDB intstance.
+
+Synopsis
+--------
+
+``edgedb project deinit [options]``
+
+Options
+-------
+
+``--project-dir=<dir>``
+  Specifies a project root directory explicitly.  If not specified, the project
+  directory is detected as described in the "Project detection" section above.
+
+``--destroy-server-instance, -D``
+  If specified, the associated EdgeDB instance is destroyed by running
```

Comment:
What is the purpose of deinit without destroy? I guess it will just leave danging instances where users expect them to be removed.

Reply:
It is mainly to fix unintentional linking.  If you picked a wrong instance by mistake, this will allow you to unlink it.